contracts/multisig-prover/src/migrate.rs

```diff
@@ -0,0 +1,143 @@
+use axelar_wasm_std::{hash::Hash, FnExt, MajorityThreshold};
+use cosmwasm_schema::cw_serde;
+use cosmwasm_std::{from_slice, Addr, DepsMut, Response, Uint256};
+use multisig::key::KeyType;
+use router_api::ChainName;
+
+use crate::{
+    encoding::Encoder,
+    state::{Config, CONFIG},
+};
+
+#[cw_serde]
+pub struct OldConfig {
+    pub admin: Addr,
+    pub governance: Addr,
+    pub gateway: Addr,
+    pub multisig: Addr,
+    pub coordinator: Addr,
+    pub service_registry: Addr,
+    pub voting_verifier: Addr,
+    pub destination_chain_id: Uint256,
+    pub signing_threshold: MajorityThreshold,
+    pub service_name: String,
+    pub chain_name: ChainName,
+    pub worker_set_diff_threshold: u32,
+    pub encoder: Encoder,
+    pub key_type: KeyType,
+}
+
+impl OldConfig {
+    pub fn migrate(self, domain_separator: Hash) -> Config {
+        Config {
+            domain_separator,
+            admin: self.admin,
+            governance: self.governance,
+            gateway: self.gateway,
+            multisig: self.multisig,
+            coordinator: self.coordinator,
+            service_registry: self.service_registry,
+            voting_verifier: self.voting_verifier,
+            destination_chain_id: self.destination_chain_id,
+            signing_threshold: self.signing_threshold,
+            service_name: self.service_name,
+            chain_name: self.chain_name,
+            worker_set_diff_threshold: self.worker_set_diff_threshold,
+            encoder: self.encoder,
+            key_type: self.key_type,
```

Comment:
Can't you just write `..self` for all of this?

Reply:
only works for exactly the same struct

Reply:
ah ok